src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelBindingContextDisposable.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    /// <summary>
+    /// Return value of <see cref="IModelBindingContext.PushContext"/>. Should be disposed
+    /// by caller when child binding context state should be popped off of 
+    /// the <see cref="IModelBindingContext"/>.
+    /// </summary>
+    public struct ModelBindingContextDisposable : IDisposable
```

Comment:
Seems like a lot of ceremony for the less-common "non-trivial .. return flow" cases. That is, I can see the benefits where it's needed but am not a fan of the increased nesting where it wraps the trivial cases. Let's chock this up to a style thing.